torch/csrc/jit/pybind_utils.h

```diff
@@ -147,10 +147,31 @@ inline IValue toIValue(py::handle obj, const TypePtr& type) {
       case TypeKind::ListType: {
         const auto& elem_type = type->expect<ListType>()->getElementType();
         switch(elem_type->kind()) {
+          //allows single int/float to be broadcasted to a fixed size list
           case TypeKind::IntType:
-            return py::cast<std::vector<int64_t>>(obj);
+            if (!N) {
+              return py::cast<std::vector<int64_t>>(obj);
+            } else if (!py::isinstance<py::int_>(obj)) {
+              auto vec = py::cast<std::vector<int64_t>>(obj);
+              JIT_ASSERTM(static_cast<int32_t>(vec.size()) == *N, "Input fixed length list has length ", vec.size(), " expected ", *N);
+              return vec;
+            } else {
+              double value = py::cast<int64_t>(obj);
+              std::vector<double> repeated(*N, value);
+              return repeated;
+            }
           case TypeKind::FloatType:
-            return py::cast<std::vector<double>>(obj);
+            if (!N) {
```

Comment:
Since the return type is different wouldn't you need a templated function? That seemed like overkill for the scenario. Or what would be another approach?

Reply:
I think it is fine if the middle case gets dropped. At its originally length a template function would have been better.